src/Compilers/CSharp/Test/Semantic/Semantics/LocalFunctionTests.cs

```diff
@@ -405,6 +405,39 @@ static void M()
             Assert.Equal(attrType, info.Symbol);
         }
 
+        [Fact]
+        public void LocalFunctionAttribute_SpeculativeSemanticModel()
+        {
+            var text = @"
+using System;
+
+class A : Attribute { }
+
+class C
+{
+    void M()
+    {
+        void local1() { }
+    }
+}
+";
+            var tree = SyntaxFactory.ParseSyntaxTree(text, TestOptions.RegularPreview);
+            var compilation = CreateCompilation(tree);
+            var model = compilation.GetSemanticModel(tree);
+
+            var newTree = SyntaxFactory.ParseSyntaxTree(text.Replace("void local1", "[A] void local1"), TestOptions.RegularPreview);
+            var localFunctionSyntax = newTree.GetRoot()
+                .DescendantNodes()
+                .OfType<LocalFunctionStatementSyntax>()
+                .Single();
+            var method = newTree.GetRoot().DescendantNodes().OfType<MethodDeclarationSyntax>().Single();
+            Assert.True(model.TryGetSpeculativeSemanticModelForMethodBody(method.Body.SpanStart, method, out var speculativeModel));
+
+            var symbol = speculativeModel.GetDeclaredSymbol(localFunctionSyntax).GetSymbol<LocalFunctionSymbol>();
+            var attrs = symbol.GetAttributes();
+            Assert.Equal(new[] { "A" }, GetAttributeNames(attrs));
```

Comment:
What's the underlying symbol here. Do we still have the type name/constructor name problem?

Reply:
Yes. Will fix that separately.